lib/client/api.go

```diff
@@ -1483,3 +1520,38 @@ func ParsePortForwardSpec(spec []string) (ports ForwardedPorts, err error) {
 	}
 	return ports, nil
 }
+
+// ToStringSpec() returns the same string spec which can be parsed by ParseDynamicPortForwardSpec
+func (fp DynamicForwardedPorts) ToStringSpec() (retval []string) {
+	for _, p := range fp {
+		retval = append(retval, p.ToString())
+	}
+	return retval
+}
+
+// ParseDynamicPortForwardSpec parses parameter to -L flag, i.e. strings like "[ip]:80:remote.host:3000"
+// The opposite of this function (spec generation) is DynamicForwardPorts.ToStringSpec()
+func ParseDynamicPortForwardSpec(spec []string) (ports DynamicForwardedPorts, err error) {
+	if len(spec) == 0 {
+		return ports, nil
+	}
+	const errTemplate = "Invalid dynamic port forwarding spec: '%s'. Could be like `local.host:8080`"
+	ports = make(DynamicForwardedPorts, len(spec), len(spec))
+
+	for i, str := range spec {
+		parts := strings.Split(str, ":")
```

Comment:
Why not use `net.SplitHostPort` here?

https://golang.org/pkg/net/#SplitHostPort